variables.tf

```diff
@@ -39,28 +33,57 @@ variable "tags" {
   default     = {}
 }
 
-variable "k8s_version" {
+variable "ssh_pvt_key_path" {
   type        = string
-  description = "EKS cluster k8s version."
-  default     = "1.25"
+  description = "SSH private key filepath."
+  validation {
+    condition     = fileexists(var.ssh_pvt_key_path)
+    error_message = "Private key does not exist. Please provide the right path or generate a key with the following command: ssh-keygen -q -P '' -t rsa -b 4096 -m PEM -f domino.pem"
+  }
 }
 
-variable "public_cidr_network_bits" {
-  type        = number
-  description = "Number of network bits to allocate to the public subnet. i.e /27 -> 32 IPs."
-  default     = 27
-}
+variable "eks" {
+  description = <<EOF
+    k8s_version = "EKS cluster k8s version."
+    kubeconfig = {
+      extra_args = "Optional extra args when generating kubeconfig."
+      path       = "Fully qualified path name to write the kubeconfig file."
+    }
+    public_access = {
+      enabled = "Enable EKS API public endpoint."
+      cidrs   = "List of CIDR ranges permitted for accessing the EKS public endpoint."
+    }
+    "Custom role maps for aws auth configmap"
+    custom_role_maps = {
+      rolearn = string
+      username = string
+      groups = list(string)
+    }
+    master_role_names = "IAM role names to be added as masters in eks."
+    cluster_addons = "EKS cluster addons. vpc-cni is installed separately."
+  EOF
+
+  type = object({
+    k8s_version = optional(string)
+    kubeconfig = optional(object({
+      extra_args = optional(string)
+      path       = optional(string)
+    }))
+    public_access = optional(object({
+      enabled = optional(bool)
+      cidrs   = optional(list(string))
+    }))
+    custom_role_maps = optional(list(object({
+      rolearn  = string
+      username = string
+      groups   = list(string)
+    })))
+    master_role_names = optional(list(string))
+    cluster_addons    = optional(list(string))
```

Comment:
not sure having them all optional makes sense, e.g., public access is either true or false.
regardless, if they are optional, what is the default behavior?

Reply:
Where possible, defaults were moved to the submodules, i.e
 https://github.com/dominodatalab/terraform-aws-eks/blob/427b677d340da19b41ba65bd67046cba04f85f35/submodules/eks/variables.tf#L141

Reply:
so now you are making a person dig to figure out if its safe to default or not.

Reply:
thats true for the passthrough vars, bastion, eks, storage and nw.